packages/@aws-cdk/aws-lambda-python/test/integ.function.requirements.removed.ts

```diff
@@ -0,0 +1,65 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { Runtime } from '@aws-cdk/aws-lambda';
+import { App, CfnOutput, Construct, Stack, StackProps } from '@aws-cdk/core';
+import * as lambda from '../lib';
+
+/*
+ * Stack verification steps:
+ * aws lambda invoke --function-name <function name> --invocation-type Event --payload $(base64 <<<'"OK"') response.json
+ */
+
+class TestStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    // Create a working directory for messing around with requirements.txt
+    const workDir = fs.mkdtempSync(path.join(os.tmpdir(), 'cdk-integ'));
+    fs.copyFileSync(path.join(__dirname, 'lambda-handler-requirements', 'index.py'), path.join(workDir, 'index.py'));
+    const requirementsTxtPath = path.join(workDir, 'requirements.txt');
+
+    // Write a requirements.txt with an extraneous dependency (colorama)
+    const beforeDeps = 'requests==2.23.0\npillow==6.2.2\ncolorama==0.4.3\n';
+    fs.writeFileSync(requirementsTxtPath, beforeDeps);
+
+    const fnBefore = new lambda.PythonFunction(this, 'before', {
+      entry: workDir,
+      dependenciesLocation: lambda.DependenciesLocation.INLINE,
+      runtime: Runtime.PYTHON_2_7,
+    });
+
+    new CfnOutput(this, 'FunctionBefore', {
+      value: fnBefore.functionName,
+    });
+
+    // Then, write a requirements.txt without the extraneous dependency
+    const afterDeps = 'requests==2.23.0\npillow==6.2.2\n';
+    fs.writeFileSync(requirementsTxtPath, afterDeps);
+
+    const fnAfter = new lambda.PythonFunction(this, 'after', {
+      entry: workDir,
+      dependenciesLocation: lambda.DependenciesLocation.INLINE,
+      runtime: Runtime.PYTHON_2_7,
+    });
```

Comment:
How does the creation of two independent lambdas affect the asset output? Yes, they're pointing at the same requirements file, but wouldn't this create two different assets?

Reply:
You're right to object to the test. It's taking advantage of internal knowledge of AssetStaging. I'll write a more convincing test that synths twice. But, I need to update from master first, to resolve an unrelated aws-sdk issue. I'll mention you when it's up. :)

Reply:
Ahhh, alright. I figured there might be something I was missing. :)